job-server/src/main/scala/spark/jobserver/AkkaClusterSupervisorActor.scala

```diff
@@ -209,38 +215,85 @@ class AkkaClusterSupervisorActor(daoActor: ActorRef, dataManagerActor: ActorRef)
         cluster.down(ref.path.address)
         ref ! PoisonPill
         failureFunc(e)
+        initContextHelp(actorName, Some(ref.path.address.toString), ContextStatus.Error, Some(e))
       case Success(JobManagerActor.InitError(t)) =>
         logger.info("Failed to initialize context " + ref, t)
         cluster.down(ref.path.address)
         ref ! PoisonPill
         failureFunc(t)
+        initContextHelp(actorName, Some(ref.path.address.toString), ContextStatus.Error, Some(t))
       case Success(JobManagerActor.Initialized(ctxName, resActor)) =>
         logger.info("SparkContext {} joined", ctxName)
         contexts(ctxName) = (ref, resActor)
         context.watch(ref)
-        successFunc(ref)
+        val optEx = initContextHelp(actorName, Some(ref.path.address.toString), ContextStatus.Running, None)
```

Comment:
you could use pattern matching here too instead of the if/else:
```
case None => successFunc(ref)
case Some(e) => {
      ref ! PoisonPill
      failureFunc(e)
}
```